cpp/src/toppra/algorithm.hpp

```diff
@@ -1,6 +1,41 @@
-#ifndef ALGORITHM_HPP
-#define ALGORITHM_HPP
+#ifndef TOPPRA_ALGORITHM_HPP
+#define TOPPRA_ALGORITHM_HPP
 
+#include <toppra/constraint.hpp>
+#include <toppra/geometric_path.hpp>
+#include <toppra/solver.hpp>
+#include <toppra/toppra.hpp>
 
+namespace toppra {
+
+struct ParametrizationData {
+  Vector K, X;
+  int ret_code = -1;
+};
+
+class PathParametrizationAlgorithm {
+public:
+  PathParametrizationAlgorithm(const LinearConstraintPtrs &constraints,
+                               const GeometricPath &path)
+      : m_constraints(constraints), m_path(path){};
+
+  /** \brief Set the level of discretization used by the solver.
+   *
+   * If is zero, will attempt to detect automatically the most suitable grid.
+   */
+  void setN(int N) { m_N = N; };
+
+  ParametrizationData getParameterizationData();
```

Comment:
The function is declared but never defined.

Reply:
Yes, it's pretty much work in progress now. This function only makes senses after the full algorithm is implemented.